pkg/tagger/collectors/kubernetes_main.go

```diff
@@ -27,19 +28,20 @@ const (
 )
 
 type KubeMetadataCollector struct {
-	kubeUtil  kubelet.KubeUtilInterface
-	apiClient *apiserver.APIClient
-	infoOut   chan<- []*TagInfo
-	dcaClient clusteragent.DCAClientInterface
+	sync.RWMutex
```

Comment:
An anonymous field makes `KubeMetadataCollector` act as a `RWMutex` in the sense that the code of other packages that is manipulating an instance of `KubeMetadataCollector` can call `Lock()` or `Unlock()` on it.

As this mutex is not aimed as being part of the public API but is rather an implementation detail of the member functions, we can turn it into a private field.

```suggestion
	m sync.RWMutex
```

Reply:
Thanks @L3n41c. Completely agree with you + [this](https://dmitri.shuralyov.com/idiomatic-go/entries/2), I somehow got under impression this is a pattern used in `tagger` but in fact it's only a thing in `autodiscovery`. Arguably many structs in this package don't even need to be exported, something to consider for another day.